backend/src/main/java/com/znane_groby/backend/controller/GraveyardController.java

```diff
@@ -0,0 +1,102 @@
+package com.znane_groby.backend.controller;
+
+import com.znane_groby.backend.model.Graveyard;
+import com.znane_groby.backend.repository.CityRepository;
+import com.znane_groby.backend.repository.GraveyardRepository;
+import com.znane_groby.backend.util.ItemNotFoundException;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.hateoas.CollectionModel;
+import org.springframework.hateoas.EntityModel;
+import org.springframework.hateoas.IanaLinkRelations;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import static org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.linkTo;
+import static org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.methodOn;
+
+@Slf4j
+@RestController
+public class GraveyardController {
+    private final GraveyardRepository repository;
+    private final CityRepository cityRepository;
+    private final GraveyardResourceAssembler assembler;
+
+    GraveyardController(GraveyardRepository repository, GraveyardResourceAssembler assembler,CityRepository cityRepository) {
+        this.assembler = assembler;
+        this.repository = repository;
+        this.cityRepository = cityRepository;
+    }
+
+    // Aggregate root
+    @GetMapping("/graveyards")
+    CollectionModel<EntityModel<Graveyard>> all() {
+
+        List<EntityModel<Graveyard>> graveyards = StreamSupport.stream(repository.findAll().spliterator(), false)
+                .map(assembler::toModel)
+                .collect(Collectors.toList());
+        return new CollectionModel<>(graveyards,
+                linkTo(methodOn(GraveyardController.class).all()).withSelfRel());
+    }
+
+    // Single item
+    @GetMapping("/graveyards/{id}")
+    EntityModel<Graveyard> one(@PathVariable Long id) {
+        Graveyard graveyard = repository.findById(id)
+                .orElseThrow(() -> new ItemNotFoundException("/graveyards/", id));
+        log.info("graveyard.getCity() " + graveyard.getCity());
```

Comment:
Jak logujesz, zwłaszcza na levelu info, to dawaj trochę więcej kontekstu. Potem zapomnisz i będą ci latać bezsensowne wpisy w logu, które nic nie mówią. Na przykład można dać tak:
```java
log.info("Graveyard controller:: get graveyard, id={}, got result: {}", id, graveyard);
```
A jeśli chcesz debugować to używaj levelu `debug` lub `trace`.

Reply:
oki